docs/wiki/TestingDesign.md

```diff
@@ -84,108 +83,90 @@ Currently the list of the parameter file used to test the module is hardcoded in
+      >**Note**: This resource has a global scope name.
+  6. Route table: This resource is leveraged by the virtual network subnet dedicated to test SQL managed instance.
+  7. Network watcher: This resource is leveraged by the NSG flow logs resource.
+  8. Shared image gallery: This resource is leveraged by the shared image definition and image template resources.
+  9. Action group: This resource is leveraged by activity log alert and metric alert resources.
+  10. Application security group: This resource is leveraged by the network security group resource.
+  11. Application service plan: This resource is leveraged by the function app and web app resources.
+
+**Third level resources**: This group of resources has a dependency on one or more resources in the group above. Resources in this group can be deployed in parallel.
+
+  12. Event hub: This resource is depending on the event hub namespace deployed above and leveraged by all resources supporting diagnostic settings on an event hub.
+  13. Role assignment: This resource assigns the '_Contributor_' role on the subscription to the user assigned identity deployed as part of the group above. This is needed by the image template deployment.
+  14. Shared image definition: This resource is depending on the shared image gallery deployed above and leveraged by the image template resource.
+
+**Fourth level resources**: All resources in this group support monitoring. Hence they have a dependency on the storage account, log analytics workspace and event hub deployed in the groups above. Resources in this group can be deployed in parallel.
+
+  15. Key vault: This resource is leveraged by all resources requiring access to a key vault key, secret and/or certificate, i.e. application gateway, azure NetApp file, azure SQL server, disk encryption set, machine learning service, private endpoint, SQL managed instance, virtual machine, virtual machine scale set, virtual network gateway connection.
+      >**Note**: This resource has a global scope name.
+  16. Network Security Groups: This resource is leveraged by different virtual network subnets. Multiple instances are deployed:
+      - '_adp-sxx-az-nsg-weu-x-apgw_': NSG with required network security rules to be leveraged by the application gateway subnet.
+      - '_adp-sxx-az-nsg-weu-x-ase_': NSG with required network security rules to be leveraged by the app service environment subnet.
+      - '_adp-sxx-az-nsg-weu-x-bastion_': NSG with required network security rules to be leveraged by the bastion host subnet.
+      - '_adp-sxx-az-nsg-weu-x-sqlmi_': NSG with required network security rules to be leveraged by the sql managed instance subnet.
+      - '_adp-sxx-az-nsg-weu-x-001_': default NSG leveraged by all other subnets.
+  17. Public IP addresses: Multiple instances are deployed:
+      - '_adp-sxx-az-pip-weu-x-apgw_': Leveraged by the application gateway resource.
+      - '_adp-sxx-az-pip-weu-x-bas_': Leveraged by the bastion host resource.
```

Comment:
This applies to all resource references: For readability we should probably encapsualte the resource references like this:
```markdown
- '_adp-sxx-az-pip-weu-x-bas_': Leveraged by the [bastion host] resource.
```

Especially when we reference multiple resources it is not that easy to directly see what resources are important

Reply:
done